common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/SearchBar.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.mobile.android.ui.view
+
+import android.content.Context
+import android.util.AttributeSet
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.widget.doOnTextChanged
+import com.duckduckgo.mobile.android.R
+import com.duckduckgo.mobile.android.databinding.ViewSearchBarBinding
+import com.duckduckgo.mobile.android.ui.viewbinding.viewBinding
+
+interface SearchBar {
+    fun onAction(actionHandler: (Action) -> Unit)
+    fun handle(event: Event)
+
+    enum class Event {
+        DismissSearchBar,
+        ShowSearchBar
+    }
+
+    sealed class Action {
+        object PerformUpAction : Action()
+        data class PerformSearch(val searchText: String) : Action()
+    }
+}
+
+class SearchBarView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : ConstraintLayout(context, attrs, defStyleAttr), SearchBar {
+    private val binding: ViewSearchBarBinding by viewBinding()
+
+    init {
+        val styledAttributes = context.obtainStyledAttributes(attrs, R.styleable.SearchBarView)
+        binding.root.background = styledAttributes.getDrawable(R.styleable.SearchBarView_android_background)
+        binding.omnibarTextInput.hint = styledAttributes.getString(R.styleable.SearchBarView_searchHint)
+        binding.clearTextButton.contentDescription = styledAttributes.getString(R.styleable.SearchBarView_clearActionContentDescription)
```

Comment:
How would you feel about having a default description? Here or in the style definition, but this is one of the fields that we almost always forget to add

Reply:
The string can live in `string-untranslated` for now and it will be translated once we order a new round

Reply:
Agreed that we will put the default strings in strings.xml in common-ui.